captum/optim/_param/image/images.py

```diff
@@ -1,538 +1,484 @@
+        return torch._tensor_str._add_suffixes(
+            prefix + tensor_str, suffixes, indent, force_newline=self.is_sparse
+        )
+
+    @classmethod
+    def __torch_function__(
+        cls: Type["ImageTensor"],
+        func: Callable,
+        types: List[Type[torch.Tensor]],
+        args: Tuple = (),
+        kwargs: dict = None,
+    ) -> torch.Tensor:
+        if kwargs is None:
+            kwargs = {}
+        return super().__torch_function__(func, types, args, kwargs)
+
+    def show(
+        self, figsize: Optional[Tuple[int, int]] = None, scale: float = 255.0
+    ) -> None:
+        show(self, figsize=figsize, scale=scale)
+
+    def export(self, filename: str, scale: float = 255.0) -> None:
+        save_tensor_as_image(self, filename=filename, scale=scale)
+
+
+class InputParameterization(torch.nn.Module):
+    def forward(self) -> torch.Tensor:
+        raise NotImplementedError
+
+
+class ImageParameterization(InputParameterization):
+    pass
+
+
+class FFTImage(ImageParameterization):
+    """Parameterize an image using inverse real 2D FFT"""
+
+    def __init__(
+        self,
+        size: Tuple[int, int] = None,
+        channels: int = 3,
+        batch: int = 1,
+        init: Optional[torch.Tensor] = None,
+    ) -> None:
+        super().__init__()
+        if init is None:
+            assert len(size) == 2
+            self.size = size
+        else:
+            assert init.dim() == 3 or init.dim() == 4
+            if init.dim() == 3:
+                init = init.unsqueeze(0)
+            self.size = (init.size(2), init.size(3))
+        self.torch_rfft, self.torch_irfft, self.torch_fft_freq = self.get_fft_funcs()
+
+        frequencies = FFTImage.rfft2d_freqs(*self.size)
+        scale = 1.0 / torch.max(
+            frequencies,
+            torch.full_like(frequencies, 1.0 / (max(self.size[0], self.size[1]))),
+        )
+        scale = scale * ((self.size[0] * self.size[1]) ** (1 / 2))
+        spectrum_scale = scale[None, :, :, None]
+        self.register_buffer("spectrum_scale", spectrum_scale)
+
+        if init is None:
+            coeffs_shape = (
+                batch,
+                channels,
+                self.size[0],
+                self.size[1] // 2 + 1,
+                2,
+            )
+            random_coeffs = torch.randn(
+                coeffs_shape
+            )  # names=["C", "H_f", "W_f", "complex"]
+            fourier_coeffs = random_coeffs / 50
+        else:
+            fourier_coeffs = self.torch_rfft(init) / spectrum_scale
+
+        self.fourier_coeffs = nn.Parameter(fourier_coeffs)
+
+    @staticmethod
+    def rfft2d_freqs(height: int, width: int) -> torch.Tensor:
+        """Computes 2D spectrum frequencies."""
+        fy = FFTImage.pytorch_fftfreq(height)[:, None]
+        # on odd input dimensions we need to keep one additional frequency
+        wadd = 2 if width % 2 == 1 else 1
+        fx = FFTImage.pytorch_fftfreq(width)[: width // 2 + wadd]
```

Comment:
nit: Both instances of `FFTImage.pytorch_fftfreq` should be changed to `self.torch_fftfreq`